### PR TITLE
Delete old artifacts starting CI and keep .tar.gz files

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
This PR modifies the CI script:
  - On starting up, old artifacts are deleted;
  - We don't delete the `.tar.gz` files created during the CI run.

These two things are convenient for running local builds.

### CODE REVIEW

File: CI.hs [hunk 1]
```diff
@@ -86,15 +93,21 @@ main = do
...
+      sDistCreateExtract target = do
         cmd "stack sdist ghc --tar-dir=."
-        [tarball] <- filter (isExtensionOf ".tar.gz") <$> getDirectoryContents "."
-        cmd $ "tar -xf " ++ tarball
-        return tarball
+        cmd $ "tar -xvf " ++ target ++ ".tar.gz"
+
+      rmFiles :: [FilePath] -> IO ()
+      rmFiles fs = forM_ fs $
+        \f -> removeFile f `catchIOError` (const  $ return ())
```
Comment (rohanjr) [inline CI.hs]: Can you check if the file exists and then remove it instead of catching the IO exception?

File: CI.hs [hunk 2]
```diff
@@ -86,15 +93,21 @@ main = do
...
+
+      rmDirs :: [FilePath] -> IO ()
+      rmDirs ds = forM_ ds $
+        \d -> removeDirectoryRecursive d `catchIOError` (const $ return ())
```
Comment (rohanjr) [inline CI.hs]: Same here

File: CI.hs (hunk 1 above)
Comment (shayne-fletcher-da) [reply CI.hs]: It's not worth it. The intent is just try and remove it and ignore any failure.

File: CI.hs (hunk 2 above)
Comment (shayne-fletcher-da) [reply CI.hs]: As above.

**Pull request merged**